tests/integration/test_charm.py

```diff
@@ -281,6 +281,103 @@ async def test_seldon_deployment(ops_test: OpsTest):
     assert response["meta"] == {}
 
 
+@pytest.mark.parametrize(
+    # server_config - server configuration file
+    # url - model prediction URL
+    # req_data - data to put into request
+    # resp_data - data expected in response
+    "server_config, url, req_data, resp_data",
+    [
+        (
+            "sklearn.yaml",
+            "api/v1.0/predictions",
+            {"data": {"ndarray": [[1, 2, 3, 4]]}},
+            {
+                "data": {
+                    "names": ["t:0", "t:1", "t:2"],
+                    "ndarray": [[0.0006985194531162835, 0.00366803903943666, 0.995633441507447]],
+                },
+                "meta": {"requestPath": {"classifier": "seldonio/sklearnserver:1.15.0"}},
+            },
+        ),
+        (
+            "sklearn-v2.yaml",
+            "v2/models/classifier/infer",
+            {
+                "inputs": [
+                    {
+                        "name": "predict",
+                        "shape": [1, 4],
+                        "datatype": "FP32",
+                        "data": [[1, 2, 3, 4]],
+                    },
+                ]
+            },
+            {
+                "model_name": "classifier",
+                "model_version": "v1",
+                "id": None,  # id needs to be reset in response
+                "parameters": {"content_type": None, "headers": None},
+                "outputs": [
+                    {
+                        "name": "predict",
+                        "shape": [1, 1],
+                        "datatype": "INT64",
+                        "parameters": None,
+                        "data": [2],
+                    }
+                ],
+            },
+        ),
+    ],
+)
+@pytest.mark.asyncio
+async def test_seldon_server(ops_test: OpsTest, server_config, url, req_data, resp_data):
+    """Test Seldon server scenario."""
```

Comment:
can we be more specific on what is tested?

Reply:
There are multiple Seldon servers defined in [configmap](https://github.com/canonical/seldon-core-operator/blob/main/src/templates/configmap.yaml.j2) under `predictor_servers`. I will rename it to that.